### PR TITLE
BO: add  a null check before passing it to checking currencies

### PR DESCRIPTION
**Pull request by oshanrube**

### CODE REVIEW

File: ps_checkpayment.php [hunk 1]
```diff
@@ -69,7 +69,7 @@ public function __construct()
         if ((!isset($this->checkName) || !isset($this->address) || empty($this->checkName) || empty($this->address))) {
             $this->warning = $this->trans('The "Payee" and "Address" fields must be configured before using this module.', array(), 'Modules.Checkpayment.Admin');
         }
-        if (!count(Currency::checkPaymentCurrencies($this->id))) {
+        if ($this->id === null || !count(Currency::checkPaymentCurrencies($this->id))) {
```
Comment (Quetzacoalt91) [inline ps_checkpayment.php]: I guess this message should be displayed only when the module is installed. It should pass the test only when `id` is different from null

Comment (Quetzacoalt91) [reply ps_checkpayment.php]:
```suggestion
        if ($this->id && !count(Currency::checkPaymentCurrencies($this->id))) {
```

**Comment by Progi1984** [comment]
@oshanrube Hi have you seen @Quetzacoalt91's review ? Thanks

**Comment by matks** [comment]
Hi,

Since we had no news for almost 1 year, I will follow [our guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/#stale-pull-requests) and I'll close this pull request as it seems stale. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**